File: engine/engine-model/src/main/scala/tools/packet/movement/MoveSummon.scala
package tools.packet.movement

import java.awt.Point

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MoveSummon(private var _characterId: Int, private var _objectId: Int, private var _startPosition: Point, private var _movementList: java.util.List[java.lang.Byte]) extends PacketInput {
  def characterId: Int = _characterId

  def objectId: Int = _objectId

  def startPosition: Point = _startPosition

  def movementList: java.util.List[java.lang.Byte] = _movementList

  override def opcode(): SendOpcode = SendOpcode.MOVE_SUMMON
}

File: engine/engine-model/src/main/java/tools/packet/movement/MoveSummon.java
package tools.packet.movement;

import java.awt.Point;
import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MoveSummon(Integer characterId, Integer objectId, Point startPosition,
                         List<Byte> movementList) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MOVE_SUMMON;
   }
}